test/differencer/rulesProcessor.test.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { NodeDiff } from "../../src/diffTool/jsonDiff";
+import { applyRules } from "../../src/diffTool/rulesProcessor";
+import * as chai from "chai";
+import path from "path";
+import fs from "fs-extra";
+import tmp from "tmp";
+import sinon from "sinon";
+import { ramlToolLogger } from "../../src";
+import chaiAsPromised from "chai-as-promised";
+
+/* eslint-disable @typescript-eslint/no-use-before-define*/
+
+const expect = chai.expect;
+const rulesPath = path.join(__dirname, "..", "..", "diffRules", "rules.json");
+let loggerSpy;
+before(() => {
+  chai.should();
+  chai.use(chaiAsPromised);
+  loggerSpy = sinon.spy(ramlToolLogger, "info");
+});
+after(() => {
+  sinon.reset();
+});
+
+describe("Test that rules application exits with relevant message when there are no differences", () => {
```

Comment:
The application isn't exiting in these tests, a function is returning a value and a message is being logged.

Reply:
it is the applyRules function that exits not the application. I will change the wording